features/cellular/easy_cellular/EasyCellularConnection.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef EASY_CELLULAR_CONNECTION_H
+#define EASY_CELLULAR_CONNECTION_H
+
+#include "CellularConnectionUtil.h"
+#ifdef CELLULAR_DEVICE
+ 
+#include "netsocket/CellularBase.h"
+
+namespace mbed {
+    
+/** EasyCellularConnection class
+ *
+ *  Simplified adapter for cellular connection
+ */
+class EasyCellularConnection: public CellularBase {
+
+public:
+    EasyCellularConnection();
+    virtual ~EasyCellularConnection();
+
+public:
+    /** Create cellular power and start dispatcher
+     *  @remark Must be called before any other methods
+     *  @return see nsapi_error_t, 0 on success
+     */
+    nsapi_error_t init();
```

Comment:
`NetworkInterface` objects are generally self-initialising.   Initialisation that require power and/or cause errors, so you don't want to do them in your constructor, can normally be deferred to first `connect`.

(Ideally `disconnect` would be a full power shutdown, so whatever you do in `init` would always happen on every new `connect`).

Doing that would be part of making this a transparent `OnboardCellularInterface` provider.

Reply:
init is no longer needed to call from application